src/frontend/middleware.go

```diff
@@ -68,6 +70,7 @@ func (lh *logHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	if v, ok := r.Context().Value(ctxKeySessionID{}).(string); ok {
 		log = log.WithField("session", v)
 	}
+	log = withTraceContext(log, r)
```

Comment:
It is good that we can do this in the middleware. This applies to all incoming HTTP requests, right?

Reply:
Right. it's straight forward for this kind of services